BEPs/BEP-362.md

```diff
@@ -0,0 +1,115 @@
+  - [1. Summary](#1-summary)
+  - [2. Motivation](#2-motivation)
+  - [3. Status](#3-status)
+  - [4. Specification](#4-specification)
+    - [4.1 Create and update bucket](#41-create-and-update-bucket)
+    - [4.2 Set the flow rate limit for a bucket](#42-set-the-flow-rate-limit-for-a-bucket)
+    - [4.3 Update bucket payment address](#43-update-bucket-payment-address)
+  - [5. License](#5-license)
+
+## 1. Summary
+The BEP introduces a storage fee paymaster solution for sponsors to cover storage costs on Greenfield, making it easier 
+for regular users to utilize the platform. Currently, users need to pay the storage fee for the buckets by themselves, 
+so they need to hold BNB in Greenfield and understand the charging mechanism. Allowing the sponsors to pay the storage 
+fee for the users will make it easier to use Greenfield.
+
+## 2. Motivation
+
+Objects stored on Greenfield incur storage fees, with additional data access incurring read traffic fees. 
+Greenfield employs stream payment to charge for storage and read traffic. For regular users, grasping the concept of 
+stream payment and knowing when to renew can be challenging. Users need to know what is the stream payment and they 
+need to calculate the cost of the BNB according to the price of storage which is confusing for normal users. 
+To address this, we aim to introduce the Paymaster approach, enabling sponsors to cover storage and read traffic costs for their users seamlessly.
+
+## 3. Status
+
+This BEP is in progress.
+
+## 4. Specification
+
+### 4.1 Create and update bucket
+When creating a bucket, users can specify a payment address for the bucket to make payments.
+
+```
+message MsgCreateBucket {  
+  string creator = 1;  
+  string bucket_name = 2;  
+  VisibilityType visibility = 3;  
+  string payment_address = 4;  
+  ...
+}
+```
+
+In previous implementations, this payment address had to be the bucket owner or be created by the bucket owner. 
+However, in the new design, we allow this payment address to be any address.
+
+Users can specify any payment address when creating the bucket. But if the payment address does not set a flow rate 
+limit(which you can refer to below sections) for the bucket, they may not use the bucket normally.
+
+Users can also update the payment address for the bucket via the `MsgUpdateBucketInfo` by specifying the payment address 
+to be replaced if they want to use another payment address after creating the bucket.
+
+We recommend setting the bucket's payment address to the sponsor's payment address for optimal convenience and flexibility.
+
+### 4.2 Set the flow rate limit for a bucket
+
+When the payment account’s owner of a bucket is different from the bucket owner, it is necessary to check if the current 
+payment flow exceeds the flow rate limit when deducting charges for streaming payments.
+
+The payment account owner can use the MsgSetBucketFlowRateLimit transaction to set the maximum payment flow size for a 
```

Comment:
change MsgSetBucketFlowRateLimit to `MsgSetBucketFlowRateLimit`, always hightlight the special wording.